homeassistant/components/sensor/kwb.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for KWB Easyfire.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.kwb/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_PORT)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pykwb==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 23
+DEFAULT_TYPE = 'tcp'
+DEFAULT_RAW = 'off'
+
+MODE_SERIAL = 0
+MODE_TCP = 1
+
+CONF_TYPE = 'type'
+CONF_RAW = 'raw'
+
+SERIAL_SCHEMA = {
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(CONF_TYPE): 'serial',
+    vol.Optional(CONF_TYPE): 'raw',
+}
+
+ETHERNET_SCHEMA = {
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.positive_int,
+    vol.Required(CONF_TYPE): 'tcp',
+    vol.Optional(CONF_TYPE): 'raw',
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the KWB component."""
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    connection_type = config.get(CONF_TYPE)
+    raw = config.get(CONF_RAW)
+
+    if raw == 'on':
+        raw = True
+
+    from pykwb import kwb
+
+    _LOGGER.info('KWB: initializing')
```

Comment:
The logger will add a kwb prefix already based on the filename, so the "KWB: " on all of these logs can be dropped. Just helps keep it a little cleaner. :)